File: gobiiproject/gobii-model/src/main/java/org/gobiiproject/gobiimodel/dto/header/DtoHeaderAuth.java
package org.gobiiproject.gobiimodel.dto.header;


import org.gobiiproject.gobiimodel.types.GobiiCropType;

import java.io.Serializable;

/**
 * Created by MrPhil on 6/18/2015.
 */
public class DtoHeaderAuth implements Serializable {


    private String userName;
    private String password;
    private String token;
    private GobiiCropType gobiiCropType;

    public void setUserName(String userName) {
        this.userName = userName;
    }

    
    public String getUserName() {
        return (this.userName);
    }

    
    public void setPassword(String password) {
        this.password = password;
    }

    
    public String getPassword() {
        return (this.password);
    }

    
    public void setToken(String token) {
        this.token = token;
    }

    
    public String getToken() {
        return (this.token);
    }

    public GobiiCropType getGobiiCropType() {
        return gobiiCropType;
    }

    public void setGobiiCropType(GobiiCropType gobiiCropType) {
        this.gobiiCropType = gobiiCropType;
    }



}//DtoHeaderResponse

File: gobiiproject/gobii-web/src/main/webapp/js/gobii_modules/model/dto-header-auth.ts

export class DtoHeaderAuth {

    public constructor(public userName:string,
                       public password:string,
                       public token:string,
                       public gobiiCropType:string ) {

        this.userName = userName;
        this.password = password;
        this.token = token;
        this.gobiiCropType=gobiiCropType;
    }


    public getToken():string {
        return this.token;
    }
    
    public getGobiiCropType():string {
        return this.gobiiCropType;
    }


    public static fromJSON(json:JSON):DtoHeaderAuth {

        return new DtoHeaderAuth(
            json['userName'],
            json['password'],
            json['token'],
            json['gobiiCropType']
        );

    }


}
